### PR TITLE
Add IP9820 power switch for control power on/off

### PR DESCRIPTION
**Pull request by SyHuangCBN**
Add Power switch for IP9820 to test power on/off.

It's power switch for 8 ports.

Support https, snmp, telnet protocol for programming.

### CODE REVIEW

File: devices/power.py [hunk 1]
```diff
@@ -271,6 +278,12 @@ def reset(self):
 #     time.delay(1)
 
 import urllib2
+from pysnmp.carrier.asyncore.dispatch import AsyncoreDispatcher
+from pysnmp.carrier.asyncore.dgram import udp
+from pyasn1.codec.ber import encoder, decoder
+from pysnmp.proto.api import v2c
+from pysnmp.entity.rfc3413.oneliner import cmdgen
+from pysnmp.proto import rfc1902
```
Comment (mattsm) [inline devices/power.py]: The automated check fail seems valid... why are we adding all of these here?

File: devices/power.py [hunk 2]
```diff
@@ -58,6 +61,10 @@ def get_power_device(ip_address, username=None, password=None, outlet=None):
         return APCPower(ip_address, outlet=outlet)
     if '<b>IP9258 Log In</b>' in data:
         return Ip9258(ip_address, outlet, username=username, password=password)
+    if 'Access Error: Unauthorized' in data:
+        print("ip_address = %s" % ip_address)
```
Comment (mattsm) [inline devices/power.py]: Is this data really unique? Can't we look for IP9820 text in the resulting output?

File: devices/power.py [hunk 3]
```diff
@@ -50,6 +51,8 @@ def get_power_device(ip_address, username=None, password=None, outlet=None):
     except Exception as e:
         print(e)
         raise Exception("\nError connecting to %s" % ip_address)
+        
+    print("data = %s" % data)
```
Comment (mattsm) [inline devices/power.py]: Please don't add your debug code in the commit.

File: devices/power.py [hunk 4]
```diff
@@ -58,6 +61,10 @@ def get_power_device(ip_address, username=None, password=None, outlet=None):
         return APCPower(ip_address, outlet=outlet)
     if '<b>IP9258 Log In</b>' in data:
         return Ip9258(ip_address, outlet, username=username, password=password)
+    if 'Access Error: Unauthorized' in data:
+        print("ip_address = %s" % ip_address)
+        print("outlet = %s" % outlet)
```
Comment (mattsm) [inline devices/power.py]: Again, please don't add debug code.

File: devices/power.py [hunk 5]
```diff
@@ -299,6 +312,123 @@ def reset(self):
...
+	self.on()
+    
+    def getPowerStatus(self):
+        oid = self.oid_Outlet + '.' + str(self.port) + '.0'
+        result = Snmp("get",self.wan_ip,"private",oid)
+        return result
+    
+    def getPowerStatusAll(self):
+        oid = self.oid_Outlet
+        result = Snmp("walk",self.wan_ip,"private",oid)
```
Comment (mattsm) [inline devices/power.py]: I see you are using some SNMP commands here actually, but does the same API that the other IP board use work instead?

File: devices/power.py [hunk 6]
```diff
@@ -299,6 +312,123 @@ def reset(self):
...
+        """Indicate whether or not to enter a case suite"""
+        if self.fall or not args:
+            return True
+        elif self.value in args: 
+            self.fall = True
+            return True
+        else:
+            return False
+            
+
```
Comment (mattsm) [inline devices/power.py]: {
           "stationname": {
                "board_type": "mv1",
                "conn_cmd": ["ssh pi@192.168.100.21 cu -l /dev/ttyUSB1 -s 115200", "ssh pi@192.168.100.21 cu -l /dev/ttyUSB0 -s 115200" ],
                "location": "unknown",
		"powerport": "cmd://on_cmd=snmpset -v 2c -c 7oWm4MC IPADDRESS .1.3.6.1.4.1.39525.11.1.2.5 i 1;off_cmd=snmpset -v 2c -c 7oWm4MC IPADDRESS .1.3.6.1.4.1.39525.11.1.2.5 i 0",
		"devices": [
			{
				"type": "debian",
				"name": "lan",
				"ipaddr": "192.168.100.21",
				"color": "blue",
				"username": "pi",
				"options": "tftpd-server"
			}
		],
		"connection_type": "local_cmd"
        }
}

Comment (SyHuangCBN) [reply devices/power.py]: We will modify to support http to control power and modify pull request.

**Pull request closed without merging**